File: 2020/day01b2.py
# Day 1 part b of the problem 
# do not repeat over the numbers already picked
# Output :
# answer : 203481432
# time   : 0.30s

import time
import sys

start = time.time()

with open('01.txt') as f:
    expenses = [int(l.strip()) for l in f]

for ii, i in enumerate(expenses):
    for jj, j in enumerate(expenses[ii:]):
        for k in expenses[ii+jj:]:
            if i + j + k == 2020:
                print(f'answer : {i * j * k}')
                print(f'time   : {(time.time() - start)*1000:.2f}ms')
                sys.exit(0)

File: 2020/day01b2.nim
# Day 1 part 2 solution in nim
# Output :
# answer : 203481432
# time   : 41 milliseconds and 505 microseconds

import lists
import sequtils
import strutils
import times

var start = now()
var expenses = initSinglyLinkedList[int]()

for line in "01.txt".lines:
    expenses.append(parseInt(line))
let expe = toSeq(expenses)
for ii,i in pairs(expe):
    for jj, j in pairs(expe[ii .. ^1]):
        for k in expe[ii+jj .. ^1]:
            if i + j + k == 2020:
                echo "answer : ", i*j*k
                echo "time   : ", now() - start